### PR TITLE
Ldap ad nodes reports

### PR DESCRIPTION
**Pull request by serbant**
AD nodes reports: https://trello.com/c/pdk3PkUK

### CODE REVIEW

**Comment by serbant** [comment]
[AD node reports](https://trello.com/c/pdk3PkUK/59-ad-node-reports)

File: ldap_probe/tasks.py [hunk 1]
```diff
@@ -352,8 +456,8 @@ def dispatch_ldap_error_report(**time_delta_args):
             If not present, the method will use the period defined by the
             :class:`citrus_borg.dynamic_preferences_registry.LdapReportPeriod`
             user preference
-    :returns: information about the arguments used to call the tass and the
-        result of :meth:`ssl_cert_tracker.lib.Email.send`
+    :returns: information about the arguments used to call the task and the
+        result of :meth:`ssl_cert_tracker.lib.Email.send'
```
Comment (DanielBustoPHSA) [inline ldap_probe/tasks.py]:
```suggestion
        result of :meth:`ssl_cert_tracker.lib.Email.send`
```

File: ldap_probe/ad_probe.py [hunk 2]
```diff
@@ -320,6 +320,7 @@ def _diagnose_network(self, err):
 
 
         """
+        self.failed = True
```
[the remark this answers is not in the document]
Comment (serbant) [reply ldap_probe/ad_probe.py]: ahhh, but there is another location where `_diagnose_network` is invoked where `self.failed = True` is not called.
valid problem, i will fix it on my active branch.

File: ldap_probe/models.py [hunk 3]
```diff
@@ -567,6 +576,34 @@ def orion_url(self):
...
             self.node.details_url, self.get_node())
 
+    @classmethod
+    def report_bad_fqdn(cls):
+        """
+        prepare data for reports about AD nodes defined in Orion but with
+        the FQDN property missing
+        """
+        return cls.annotate_orion_url(
+            cls.objects.filter(Q(node__node_dns__isnull=True) |
```
[the remark this answers is not in the document]
Comment (serbant) [reply ldap_probe/models.py]: good point. the filter should check for empty string, not for is null. fixing on the current branch.

**Pull request merged**